app/src/main/java/com/quarkworks/apartmentgroceries/service/SyncGroceryItem.java

```diff
@@ -38,13 +48,30 @@ public void done(@Nullable JSONObject jsonObject) {
                 try {
 
                     JSONArray groceryJsonArray = jsonObject.optJSONArray(JsonKeys.RESULTS);
-
-                    for (int i = 0; i < groceryJsonArray.length(); i++) {
-                        RGroceryItem groceryItem = realm.createObject(RGroceryItem.class);
-                        groceryItem.setName(groceryJsonArray.getJSONObject(i).optString(JsonKeys.NAME));
+                    if (groceryJsonArray != null) {
+                        for (int i = 0; i < groceryJsonArray.length(); i++) {
+                            RGroceryItem groceryItem = realm.createObject(RGroceryItem.class);
+                            groceryItem.setGroceryId(groceryJsonArray.getJSONObject(i)
+                                    .getString(JsonKeys.OBJECTID));
+                            groceryItem.setName(groceryJsonArray.getJSONObject(i)
+                                    .getString(JsonKeys.NAME));
+                            groceryItem.setGroupId(groceryJsonArray.getJSONObject(i)
+                                    .getJSONObject(JsonKeys.GROUPID).getString(JsonKeys.OBJECTID));
+                            groceryItem.setGroupName(groceryJsonArray.getJSONObject(i)
+                                    .getJSONObject(JsonKeys.GROUPID).getString(JsonKeys.NAME));
+                            groceryItem.setCreatedBy(groceryJsonArray.getJSONObject(i)
```

Comment:
This would look a lot cleaner if you got your JSONObject first and then looked for the keys in it. ie)

JSONObject groceryJsonObject.getJSONObject(i);

groceryItem.setName(groceryJsonObject.getString(JsonKeys.NAME));

.....